publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -304,12 +307,45 @@ private Publication updateMetadata(UpdatePublicationRequest input,
         var customerApiClient = getCustomerApiClient();
         var customer = fetchCustomerOrFailWithBadGateway(customerApiClient, publicationUpdate.getPublisher().getId());
         validatePublication(publicationUpdate, customer);
+        injectFileUploadDetails(existingPublication, publicationUpdate, userInstance);
         setRrsOnFiles(publicationUpdate, existingPublication, customer, userInstance.getUsername(), permissionStrategy);
         upsertPublishingRequestIfNeeded(existingPublication, publicationUpdate, customer, requestInfo);
 
         return resourceService.updatePublication(publicationUpdate);
     }
 
+    private static void injectFileUploadDetails(Publication existingPublication, Publication publicationUpdate,
+                                                UserInstance userInstance) {
+        if (!existingPublication.getAssociatedArtifacts().equals(publicationUpdate.getAssociatedArtifacts())) {
+            var username = new Username(userInstance.getUsername());
+            var uploadDetails = new UploadDetails(username, Instant.now());
+
+            var originalArtifacts = existingPublication.getAssociatedArtifacts();
+            var updatedArtifacts = publicationUpdate.getAssociatedArtifacts();
+
+            for (var updatedArtifact : updatedArtifacts) {
+                if (!originalArtifacts.contains(updatedArtifact)) {
+                    if (updatedArtifact instanceof File file) {
```

Comment:
Kan dette medføre at vi oppdaterer uploadDetails for associated artefact som kun har fått endret f.eks lisens?

Reply:
Nei, for da går den ikke gjennom sjekken på linje 327. Men man trenger jo egentlig ikke å komme helt dit dersom det kun er en update på en fil. Så det vil kanskje være bedre om jeg endrer sjekken på linje 319 til å sjekke om det er lagt til nye filer (og ikke bare om **noe** har endret seg)

Reply:
Nvm, var litt naïv på den .contains()-metoden. Du har rett, en endring på fil vil passere gjennom 😬 Skal oppdatere, og legge til en test som passer sjekker dette.

Reply:
Foreslår en liten endring her:

```

    private static void injectFileUploadDetails(AssociatedArtifactList originalArtifacts,
                                                AssociatedArtifactList updatedArtifacts,
                                                UploadDetails uploadDetails) {
        if (originalArtifacts.equals(updatedArtifacts)) {
            return;
        }
        for (var updatedArtifact : updatedArtifacts) {
            if (!originalArtifacts.contains(updatedArtifact) && updatedArtifact instanceof File file) {
                updateAssociatedArtifacts(updatedArtifacts, uploadDetails, file);
            }
        }
    }

    private static void updateAssociatedArtifacts(AssociatedArtifactList updatedArtifacts,
                                                  UploadDetails uploadDetails,
                                                  File item) {
        var index = updatedArtifacts.indexOf(item);
        var updated = updateFileWithUploadDetails(item, uploadDetails);
        updatedArtifacts.set(index, updated);
    }

    private static UploadDetails extractUploadDetails(UserInstance userInstance) {
        return new UploadDetails(new Username(userInstance.getUsername()), Instant.now());
    }

    private static File updateFileWithUploadDetails(File file, UploadDetails uploadDetails) {
        return switch (file) {
            case PublishedFile publishedFile -> addUploadDetails(publishedFile, uploadDetails).buildPublishedFile();
            case UnpublishedFile unpublishedFile ->
                addUploadDetails(unpublishedFile, uploadDetails).buildUnpublishedFile();
            default -> addUploadDetails(file, uploadDetails).buildUnpublishableFile();
        };
    }

    private static Builder addUploadDetails(File file, UploadDetails uploadDetails) {
        return file.copy().withUploadDetails(uploadDetails);
    }
```
Tror at jeg heller ville endret signaturen til `injectFileUploadDetails` til 

```
    AssociatedArtifactList updateAssociatedArtifactList(AssociatedArtifactList originalArtifacts,
                                                AssociatedArtifactList updatedArtifacts,
                                                UploadDetails uploadDetails) {…}
```

Og oppdatert PublicationUpdate etterpå da disse muteringer er litt vanskelig å følge.